src/stream/src/executor/watermark_filter.rs

```diff
@@ -229,6 +234,19 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
                             table.insert(row);
                         }
                         table.commit(barrier.epoch).await?;
+
+                        if idle_input {
```

Comment:
I'm unsure if it's reasonable to use a single checkpoint as the threshold for determining an idle source, considering that remote reads are involved. Let's wait and see.